Upgrade: fix stuck in the pre-drained state

bk201 opened the pull request:
**IMPORTANT: Please do not create a Pull Request without creating an issue first.**

**Problem:**
We have spotted some situations instance-manager pod can't be drained out of a node:

- Longhorn is being upgraded, and there are 2 instance-manager-r pods on a node.
- A node is just drained and comes back; if we drain the next node, there is a high probability the next node is stuck.

**Solution:**
<!-- Example: When "Adding a function to do X", explain why it is necessary to have a way to do X. -->
- Wait Longhorn finish upgrading instance-manager-r pods
- Wait for replicas ready before draining a node

**Related Issue:**
https://github.com/harvester/harvester/issues/2741

**Test plan:**
<!-- Make sure tests pass on the Circle CI. -->
Build and upgrade from v1.0.3.
- Test 1-node cluster upgrade. (changes in this PR not run )
- Test 2-node cluster upgrade.
- Test 3-node cluster upgrade.

bk201 committed:
Upgrade: wait LH finish upgrading instance-manager-r pods

Signed-off-by: Kiefer Chang <kiefer.chang@suse.com>

bk201 committed:
Upgrade: wait for replicas ready before draining a node

Signed-off-by: Kiefer Chang <kiefer.chang@suse.com>

Vicente-Cheng reviewed (approved):
LGTM, and tested it. After the upgrade, VM looks fine and the data is correct.
Thanks!

bk201 committed:
upgrade-node: handle 1x replica in two-nodes situation

It's possible user might configure 1x replica volumes. Wait
for only 1 replica to be ready in this situation.

Signed-off-by: Kiefer Chang <kiefer.chang@suse.com>

starbops reviewed (approved):
LGTM! Tested on 1/2/3/4-node upgrades.
